### PR TITLE
[ADD] module unshared_currency

### PR DESCRIPTION
**Pull request by bealdav**
Comes from this PR: split in 2 modules
https://github.com/OCA/multi-company/pull/1

### CODE REVIEW

**Comment by bealdav** [comment]
Failed solved by this PR https://github.com/OCA/multi-company/pull/6.
Is somebody with rights could rebuild this PR. How to do it ?
Thanks

**Comment by pedrobaeza** [comment]
You have to rebase your PR.

**Comment by bealdav** [comment]
I do not understand how runbot works, and how see what failed. :scream_cat:

**Comment by pedrobaeza** [comment]
You don't have to take into account runbot. It's not working properly. Just take Travis information.

**Comment by pedrobaeza** [comment]
A better name would be _base_currency_multicompany_

**Comment by bealdav** [comment]
There 2 ways to use currencies in multi-company environnement:
- unshare currencies: the purpose of this module
- share currencies: see https://github.com/odoo/odoo/issues/641

This module deal with unshared currencies only.

File: unshared_currency_multicompany/currency.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    _inherit = 'res.currency'
+
+    def name_get(self, cr, uid, ids, context=None):
+        """ If currencies are not shared and they have the same name,
+            then display have to distinguish them"""
+        result = []
+        for currency in self.browse(cr, uid, ids, context):
+            cpny_info = ''
+            if currency.company_id:
+                cpny_info = ' (%s)' % currency.company_id.name
```
Comment (rvalyi) [inline unshared_currency_multicompany/currency.py]: sorry if that's a dumb question, but why use that 1st interpolation here (' (%s)' %) ?
can you not just do:
result.append((currency.id, '%s %s' % (currency.name, currency.company_id.name)))
?

**Comment by rvalyi** [comment]
@pedrobaeza for me a name like base_\* is for modules which are "virtual" in the sense that they only become useful when extended by other "concrete" modules. This is not the case here as this module is already usable alone. Also the name "unshare_" is quite explicit. So the name is okay for me, what do you think?

:+1:  LGTM

**Comment by pedrobaeza** [comment]
It's more like an standard rule for names. Look at my comment in https://github.com/odoomrp/odoomrp-wip/issues/431#issuecomment-66614814, where I try to explain an homogeneus rule, but I'm not going to block this PR for this reason.

**Comment by bealdav** [comment]
Thanks @pedrobaeza and @rvalyi, could I have your position on
https://github.com/odoo/odoo/issues/641
because to know what to put in homogeneus rules on multicurrencies, we have to find the optimal position. On my side I think, currencies should be shared. EUR is EUR, if your in US, pakistan, or in Argentina. Only currency rates evolve with time

**Comment by yvaucher** [comment]
I agree with @rvalyi that `base_` prefix should be used by module which are abstract module giving base for other module as dependency.

**Comment by bealdav** [comment]
Maybe It can be merged now ?

**Comment by Viggor** [comment]
Seems good for me :+1:

**Comment by yvaucher** [comment]
:+1:

**Pull request merged**